pkg/runtime/generate.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2019 GM Cruise LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runtime
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	serializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	kubernetesscheme "k8s.io/client-go/kubernetes/scheme"
+)
+
+const indentString = "    "
+
+var out = func(format string, a ...interface{}) { fmt.Printf(format, a...) }
+
+func Generate(path string) error {
+	path, err := filepath.Abs(path)
+	if err != nil {
+		return err
+	}
+	yamlOrJSONFile, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	yamlsOrJSONs := bytes.Split(yamlOrJSONFile, []byte(`---`))
+	a := newAddonFile()
+
+	scheme := k8sruntime.NewScheme()
+	_ = kubernetesscheme.AddToScheme(scheme)
+	_ = apiextensionsv1beta1.AddToScheme(scheme) // support for CRDs
+	decode := serializer.NewCodecFactory(scheme).UniversalDeserializer().Decode
+
+	for _, yamlOrJSON := range yamlsOrJSONs {
+		if len(bytes.TrimSpace(yamlOrJSON)) == 0 {
+			continue
+		}
+		obj, _, err := decode(yamlOrJSON, nil, nil)
+		if err != nil {
+			return err
+		}
+		a.addObject(obj)
+	}
+	starlark := a.gen()
+	out("%s", starlark)
+	return nil
+}
+
+type addonFile struct {
+	pkgMap  map[string]string
+	pkgs    []string
+	objects []k8sruntime.Object
+	names   []string
+}
+
+func newAddonFile() *addonFile {
+	return &addonFile{
+		pkgMap: map[string]string{},
+	}
+}
+
+func (a *addonFile) addObject(object k8sruntime.Object) {
+	a.objects = append(a.objects, object)
+}
+
+func (a *addonFile) gen() []byte {
+	out := bytes.NewBuffer([]byte{})
+	out.WriteString("# vim: set syntax=python:\n\n")
+
+	// imports
+	kubePut := a.getKubePut(1)
```

Comment:
`getKubePut(1)` is a bit cryptic on what it means. Consider renaming it to `genKubePutWithIndent(1)`, or maybe use the option pattern so that the indent count has a default valet of 1, and we have 
```go
func (a *addonFile) genKubePut(opts ...Option) { ... }
func WithIndent(int i) Option { ... }
```
also, I think `gen*` is more descriptive than `get*` and is consistent with the parent caller function `func (a *addonFile) gen() []byte {}` here